src/messageschannel.cpp

```diff
@@ -119,18 +122,26 @@ void MatrixMessagesChannel::onPendingEventChanged(int pendingEventIndex)
     header[QStringLiteral("message-sender-id")] = QDBusVariant(m_targetId);
     header[QStringLiteral("message-type")]      = QDBusVariant(Tp::ChannelTextMessageTypeDeliveryReport);
     header[QStringLiteral("delivery-status")]   = QDBusVariant(tpDeliveryStatus);
-    header[QStringLiteral("delivery-token")]    = QDBusVariant(pendingEvent.event()->transactionId());
+    header[QStringLiteral("delivery-token")]    = QDBusVariant(pendingEvent.event()->id());
     partList << header;
 
     addReceivedMessage(partList);
 }
 
+void MatrixMessagesChannel::sendChatStateNotification(uint state)
+{
+    m_room->connection()->
+            callApi<QMatrixClient::SetTypingJob>
+            (QMatrixClient::BackgroundRequest,
+             m_connection->matrix()->user()->id(), m_room->id(), (state == Tp::ChannelChatStateComposing), 10000); // FIXME: magic number
```

Comment:
The last issue to solve are the magic numbers and then we can accept this :+1: as a big step forward :).

I would suggest to add
```
constexpr int c_repeatInterval = 5000;
```
and use it for the timer.

As for about the SetTypingJob, the timeout is optional so maybe we can drop the argument for now?

Reply:
Thank you!
I'm not sure about `c_chatStateResendInterval` apropriate place and leaved it right before `setInterval` method